src/git.rs

```diff
@@ -444,3 +494,171 @@ impl Drop for TempRepo {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use git2::{Oid, Reference};
+    use quickcheck_macros::quickcheck;
+
+    fn valid_mainlines(mainlines: &[String]) -> bool {
+        mainlines
+            .iter()
+            .all(|mainline| !mainline.contains('\u{0}') && Reference::is_valid_name(&mainline))
+    }
+
+    #[test]
+    fn is_mainline_with_default_config_only_identifies_head_branch() {
+        let project_root = env!("CARGO_MANIFEST_DIR");
+        let git = LiveGit::default(format!("{}/tests/test-repo.git", project_root)).unwrap();
+        assert_eq!(git.is_mainline("refs/heads/master").unwrap(), true);
+        assert_eq!(git.is_mainline("refs/heads/tagged-branch").unwrap(), false);
+    }
+
+    #[test]
+    fn is_mainline_with_glob_config_does_not_identify_head_branch() {
+        let project_root = env!("CARGO_MANIFEST_DIR");
+        let git = LiveGit::new(
+            format!("{}/tests/test-repo.git", project_root),
+            GitConfig {
+                mainlines: vec!["tagged-*".into()],
+            },
+        )
+        .unwrap();
+        assert_eq!(git.is_mainline("refs/heads/master").unwrap(), false);
+        assert_eq!(git.is_mainline("refs/heads/tagged-branch").unwrap(), true);
+    }
+
+    #[test]
+    fn is_mainline_with_literal_config_does_not_identify_head_branch() {
```

Comment:
So isn't glob based search for the literal "tagged-branch" equivalent to the literal search?

Reply:
For `is_mainline`, yes it is the same. I wanted an explicit test for this case because the glob behaviour on `find_new_commits` needed special logic, so it seemed like a good idea to double check.